### PR TITLE
[path_provider] Use the application ID in the application support path

### PR DESCRIPTION
**Pull request by robert-ancell**
## Description

Use the application ID in the application support path. The existing code uses the binary name, which is not necessarily unique. Application IDs were recently added to the Linux shell.

## Related Issues

https://github.com/flutter/flutter/issues/59234

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [x] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [ ] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: packages/path_provider/path_provider_linux/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+## 0.1.0
+* This release updaes getApplicationSupportPath to use the application ID instead of the executable name.
```
Comment (stuartmorgan) [inline packages/path_provider/path_provider_linux/CHANGELOG.md:2]: updates

File: packages/path_provider/path_provider_linux/CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,7 @@
...
+  * Older apps will lose their preferences as they migrate to the new app ID based path.
```
Comment (stuartmorgan) [inline packages/path_provider/path_provider_linux/CHANGELOG.md:3]: app-ID-based

Comment (stuartmorgan) [inline packages/path_provider/path_provider_linux/CHANGELOG.md:3]: It's not preferences, it's any data stored by previous versions under getApplicationSupportPath. Lot of packages use path_provider.

**Comment by ditman** [comment]
This PR seems to have broken tests in master?

```
Starting application: test_driver/path_provider_e2e.dart
Building Linux application...                                   
flutter: Observatory listening on http://127.0.0.1:40955/2sSfCrcKzAA=/
flutter: 00:00 +0: getTemporaryDirectory
flutter: 00:00 +1: getDownloadDirectory
flutter: 00:00 +2: getApplicationDocumentsDirectory
flutter: 00:00 +3: getApplicationSupportDirectory
Timed out!
```

**Comment by stuartmorgan** [comment]
@robert-ancell Is there a bug tracking re-landing this?

**Comment by robert-ancell** [comment]
> @robert-ancell Is there a bug tracking re-landing this?

I'm working on a fixed version in https://github.com/flutter/plugins/pull/3077

**Pull request merged**